### DISCUSSION TITLE
Install without halting on failure (ignore errors)

### DISCUSSION THREAD

**Discussion by thw0rted** [npm/feedback#918] [CLI]
I'm in the process of setting up a private package mirror.  I thought I had made copies of all my dependencies, but it turns out some are missing.  Now, when I `npm install`, it gets part-way through installing dependencies, finds one that isn't in my configured registry, and stops.  I upload the missing package, `npm install` again, and it fails on the next one that's missing.  Repeat over and over until I finally catch them all....

I'd like a way to invoke the command such that it tries to install every dep it knows about, even if it fails to install one or more packages along the way.  I haven't found anything like this as a flag to `npm install` (or `npm ci`), but maybe there's another approach I'm missing here?  I guess if there's no way to do this today, then I'd like to suggest a CLI flag to continue-on-error.

**Comment by ljharb** [comment]
Can’t you set up your mirror to auto-download from public npm and cache? That’s usually how private registries work.

**Reply by thw0rted** [reply]
The registry in question is not connected to the internet. I brought over a bunch of TGZ packages (downloaded with `npm pack some-package`) manually.

I actually wound up in this situation because I ran `npm publish ./some-package*.tgz` for each package (with the custom registry in my npmrc), but some of them silently failed to show up. I think it might be a bug in Nexus, but haven't had time to pursue that yet. In the meantime, I had to go through the process outlined in the OP.

**Reply by thw0rted** [reply]
Before I forget: I ran into some other weirdness during this process that might be worth a look.  I reported https://github.com/npm/cli/issues/6400 because there were some packages where `npm publish ./some-package*.tgz` cannot be used due to a `publishConfig.registry` in the TGZ's `package.json`.

~I also ran into some weirdness where `npm cache clean --force; npm cache add ./some-package*.tgz; npm cache ls some-package` does not throw any errors but also has no results.  I didn't have time to put together a repro for that one yet but I wonder if it's related to the apparent Nexus bug I described in my previous reply.~

...never mind, now I can't get `npm cache ls some-package` to show results for any files added with `npm cache add file.tgz`.  I thought I was getting results sometimes but it must have been for stuff I `npm install`'d.  `npm cache ls` lists added files as e.g. `pacote:tarball:file:yargs-15.4.1.tgz`.  I can't `npm install --offline some-cached-package`, it wants to go online and look up the package metadata.  Maybe `npm cache` just doesn't work the way I was hoping it would?
